server/graphql/v2/object/Host.js

```diff
@@ -437,6 +452,124 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          account: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          dateFrom: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics beginning from this date.',
+          },
+          dateTo: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics until this date.',
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = { HostCollectiveId: host.id, kind: 'CONTRIBUTION' };
+          const dateRange = getFilterDateRange(args.dateFrom, args.dateTo);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          if (args.account) {
+            const collectiveIds = args.account.map(async account => {
+              const accountObj = await fetchAccountWithReference(account, { throwIfMissing: true });
+              return accountObj.id;
+            });
+            await Promise.all(collectiveIds).then(collectiveIds => {
+              where.CollectiveId = { [Op.in]: collectiveIds };
+            });
```

Comment:
This seems like a need you could see in other places, worth adding a helper (eg. `fetchAccountsWithReferences`) in my opinion. This part is a personal preference.

However, the part I'm blocking on is the fetching of collectives one by one, it's necessarily expensive: try it with 10 accounts, you'll trigger 10 SQL queries.

```es6
export const fetchAccountsWithReferences = async (inputs, ...) => {
  const conditions = inputs.map(getSQLConditionFromAccountReferenceInput);
  const accounts = await models.Collective.findAll({ where: { [Op.or]: conditions } });
  if (account.length !== inputs.length && throwIfMissing) {
    ...
  }

  return accounts;
}
```

```suggestion
            const accounts = await fetchAccountsWithReferences(args.account, { ... });
            where.CollectiveId = accounts.map(a => a.id);
```

Reply:
Ah, thanks for the example. Should have caught this yesterday. Now I've refactored the code and added a new function as you suggested which just gives the collection of collectives in one SQL. Let me know if there's any other suggestions on this. 😉